Add OpenTelemetry example with Uptrace backend

vmihailenco opened the pull request:
This PR adds some docs that explain how start monitoring redis-py performance using [OpenTelemetry](https://opentelemetry.io/) and [Uptrace](https://github.com/uptrace/uptrace). Here is how the [doc](https://vmihailenco-redis-py.readthedocs.io/en/latest/opentelemetry.html) page will look like.

I've briefly discussed this on Slack with @chayim and he encouraged me to open a PR to start a discussion. So here I am :)

chayim commented:
@vmihailenco I think we should probably move these things from the docker folder (to start) into docs/examples. Perhaps even start a subsection on monitoring. WDYT?

chayim commented on docs/opentelemetry.rst:
```diff
@@ -0,0 +1,102 @@
+
+.. code-block:: shell
+
+   pip install opentelemetry-instrumentation-redis==0.35b0
```
IMHO if we tie this to a version, it's one more thing to maintain.

vmihailenco replied:
Agreed, it is a copy-paste mistake

chayim commented on docs/opentelemetry.rst:
```diff
@@ -0,0 +1,102 @@
+.. code-block:: python
+
+   # Sync client
+   client = redis.StrictRedis(host="localhost", port=6379)
```
StrictRedis is now redis - so redis.Redis?

vmihailenco replied:
Done

chayim commented on docs/opentelemetry.rst:
```diff
@@ -0,0 +1,102 @@
+   client.get("my-key")
+
+   # Async client
+   client = redis.asyncio.Redis(host="localhost", port=6379)
```
Actually, also defaults are localhost/6379

vmihailenco replied:
Yeah, but my thinking is that with explicit values users can more easily copy-paste the example and change credentials. WDYT?

vmihailenco commented:
>I think we should probably move these things from the docker folder (to start) into docs/examples. Perhaps even start a subsection on monitoring. WDYT?

A subsection is probably too much to start with, but I will move everything to docs/examples :+1:

vmihailenco commented:
>Perhaps even start a subsection on monitoring

I think I misread subsection as subfolder. I guess you mean a section in examples.rst? I will play with it.

vmihailenco commented:
@chayim PTAL. All should be fixed, but I was not able to update `examples.rst` in a meaningful manner. Perhaps you could show me an example?

vmihailenco commented:
The CI failure seems unrelated...

vmihailenco commented:
@chayim I've added the [Jupyter notebook](https://vmihailenco-redis-py.readthedocs.io/en/latest/examples/opentelemetry_api_examples.html) we've discussed and updated the title. PTAL

vmihailenco commented:
@dvora-h thanks!
